python_modules/dagster/dagster/core/execution_plan/simple_engine.py

```diff
@@ -223,20 +232,31 @@ def _execute_steps_core_loop(step_context, inputs, intermediates_manager):
     yield DagsterEvent.step_start_event(step_context)
 
     with time_execution_scope() as timer_result:
-        step_output_value_iterator = check.generator(
-            _iterate_step_output_values_within_boundary(step_context, evaluated_inputs)
+        step_output_iterator = check.generator(
+            _iterate_step_outputs_within_boundary(step_context, evaluated_inputs)
         )
-    for step_output_value in check.generator(
-        _error_check_step_output_values(step_context.step, step_output_value_iterator)
+    for step_output in check.generator(
+        _error_check_step_outputs(step_context.step, step_output_iterator)
     ):
-        yield _create_step_event(step_context, step_output_value, intermediates_manager)
+        if isinstance(step_output, StepOutputValue):
```

Comment:
ah I see the tension you were talking about now with respect to Result versus StepOutputValue. Seems like we should refactor this to the point where the type checks against Result and Materialization are at the same layer

Reply:
Ah i see you do it in the other spot too.